File: OOP_Variable/3.rb
class Cal
    def initialize(v1, v2) # 생성자(Constructor) 새로운 인스턴스가 생기면 반드시 실행됨! (=초기화)
        @v1 = v1 # 앞에 @가 붙으면 인스턴스 변수. 즉, 해당 Class 안의 모든 메소드에서 사용됨
        @v2 = v2
        # @가 붙지 않으면 지역 변수. 즉 initialize 안에서만 사용 가능
    end        

    def add()
        return @v1 + @v2
    end
    
    def subtract()
        return @v1 - @v2
    end

    def setV1(v)
        if v.is_a?(Integer)
            @v1 = v
        end
    end

    def getV1()
        return @v1
    end

end

c1 = Cal.new(10,10) #c1 라는 인스턴스
p c1.add()
p c1.subtract()
c1.setV1(20)
c1.setV1('one') #setV1의 <v.is_a?(Integer)>을 통해 'one'이라는 문자열은 무시됨
p c1.add()
p c1.getV1()

File: OOP_Variable/3.py
class Cal(object): 
    def __init__(self, v1, v2): 
        self.v1 = v1
        self.v2 = v2

    def add(self): 
        return self.v1 + self.v2
        
    def subtract(self):
        return self.v1 - self.v2

    def setV1(self, v1):
        if isinstance(v1, int): # 첫 번째 인자(v1)가 만약 int 라는 class에 해당 된다면 True, 아니면 Flase
                                # 단순히 c1.v1 = 20 를 할 경우, isinstance 처럼 입력값을 확인할 수 있는 방법이 없음 (메소드로서 작성한 이유임!)
            self.v1 = v1
            
    def getV1(self):
        return self.v1

c1 = Cal(10,10)
print(c1.add())
print(c1.subtract())
c1.setV1('one')
c1.setV1(20)
print(c1.add())
print(c1.getV1())